lab_5_scrapper/scrapper.py

```diff
@@ -120,46 +261,207 @@ def __init__(self, full_url: str, article_id: int, config: Config) -> None:
+        topic = topic_tag.get_text(strip=True)
+        self.article.topics = topic
+
+
+        date_tag = article_soup.find('div', {'class': 'article__date'})
+        date_str = date_tag.get_text(strip=True)
+        date = self.unify_date_format(date_str)
+        self.article.date = date
 
     def unify_date_format(self, date_str: str) -> datetime.datetime:
         """
         Unifies date format
         """
-        pass
+        months = {
+            "января": "january",
+            "февраля": "february",
+            "марта": "march",
+            "апреля": "april",
+            "мая": "may",
+            "июня": "june",
+            "июля": "july",
+            "августа": "august",
+            "сентября": "september",
+            "октября": "october",
+            "ноября": "november",
+            "декабря": "december"
+        }
+        for rus_month, en_month in months.items():
+            date_str = date_str.replace(rus_month, en_month)
+        if 'сегодня' in date_str:
+            today = datetime.datetime.now()
+            date_str = date_str.replace('сегодня', today.strftime('%d %B %Y'))
+        elif 'вчера' in date_str:
+            yesterday = datetime.datetime.now() - datetime.timedelta(days=1)
+            date_str = date_str.replace('вчера', yesterday.strftime('%d %B %Y'))
+        regular = re.match(r"\d{1,2} [A-Za-z]+ \d{4}, \d{1,2}:\d{2}", date_str)
+        if regular:
+            return datetime.datetime.strptime(date_str, '%d %B %Y, %H:%M')
+        return datetime.datetime.strptime(date_str, '%H:%M, %d %B %Y')
+
+
 
     def parse(self) -> Union[Article, bool, list]:
         """
         Parses each article
         """
-        pass
+        response = make_request(self.article_url, self.config)
+        article_soup = BeautifulSoup(response.text, 'html.parser')
+        self._fill_article_with_text(article_soup)
+        self._fill_article_with_meta_information(article_soup)
+        return self.article
+
 
 
 def prepare_environment(base_path: Union[Path, str]) -> None:
     """
     Creates ASSETS_PATH folder if no created and removes existing folder
     """
-    pass
+    if base_path.exists():
+        shutil.rmtree(base_path)
+    base_path.mkdir(parents=True)
+
+
+class CrawlerRecursive(Crawler):
+    """
+    Recursive Crawler implementation
+    """
+
+    def __init__(self, config: Config) -> None:
+        """
+        Initializes an instance of the Recursive Crawler class
+        """
+        super().__init__(config)
+        self.start_url = self.get_search_urls()[0]
+        self.urls = []
+        self.crawler_data_path = Path('crawler_sta.json')
+        if self.crawler_data_path.exists():
+            self.load_crawler_data()
+
+    def save_crawler_data(self) -> None:
+        """
+        Saves values of start_url and urls attributes
+        in a json file
+        """
+        info_dict = {'start_url': self.start_url, 'urls': self.urls}
```

Comment:
ok, be ready to demonstrate in all scenarios that were listed before in TG channel